wire-ios-data-model/Tests/MLS/MLSServiceTests.swift

```diff
@@ -2650,4 +2650,143 @@ class MLSServiceTests: ZMConversationTestsBase, MLSServiceDelegate {
+            XCTAssertEqual(qualifiedID, uiMOC.performAndWait { conversation.qualifiedID })
+            XCTAssertEqual(messageProtocol, .mixed)
+            XCTAssert(notificationContext === uiMOC.notificationContext)
+            updateConversationProtocolExpectation.fulfill()
+        }
+
+        let syncConversationExpectation = XCTestExpectation(description: "syncConversation must be called")
+        mockActionsProvider.syncConversationQualifiedIDContext_MockMethod = { [uiMOC] qualifiedID, notificationContext in
+            uiMOC.performAndWait {
+                XCTAssertEqual(qualifiedID, conversation.qualifiedID)
+                XCTAssert(notificationContext === uiMOC.notificationContext)
+                conversation.messageProtocol = .mixed
+            }
+            syncConversationExpectation.fulfill()
+        }
+
+        let createConversationExpectation = XCTestExpectation(description: "createConversation must be called")
+        mockCoreCrypto.mockCreateConversation = { conversationID, _, _ in
+            XCTAssertEqual(conversationID, [UInt8](mlsGroupID.data))
+            createConversationExpectation.fulfill()
+        }
+
+        let updateKeyMaterialExpectation = XCTestExpectation(description: "updateKeyMaterial must be called")
+        mockMLSActionExecutor.mockUpdateKeyMaterial = { [self] mlsGroupID in
+            XCTAssertEqual(mlsGroupID, uiMOC.performAndWait { conversation.mlsGroupID })
+            updateKeyMaterialExpectation.fulfill()
+            return []
+        }
+
+        let commitPendingProposalsExpectation = XCTestExpectation(description: "commitPendingProposals must be called")
+        mockMLSActionExecutor.mockCommitPendingProposals = { _ in
+            commitPendingProposalsExpectation.fulfill()
+            throw MLSActionExecutor.Error.noPendingProposals
+        }
+
+        // Mock claiming a key package.
+        var keyPackage: KeyPackage!
+        mockActionsProvider.claimKeyPackagesUserIDDomainExcludedSelfClientIDIn_MockMethod = { [self] userID, domain, _, _ in
+            keyPackage = createKeyPackage(userID: userID, domain: domain ?? BackendInfo.domain!)
+            return [keyPackage]
+        }
+
+        // Mock adding memebers to the conversation.
+        var addedMembers = [(invitee: [Invitee], mlsGroupID: MLSGroupID)]()
+        let updateEvent = dummyMemberJoinEvent()
+        mockMLSActionExecutor.mockAddMembers = {
+            addedMembers.append(($0, $1))
+            return [updateEvent]
+        }
+
+        // When
+        try await sut.startProteusToMLSMigration()
+
+        // Then
+        await fulfillment(
+            of: [
+                updateConversationProtocolExpectation,
+                syncConversationExpectation,
+                createConversationExpectation,
+                updateKeyMaterialExpectation,
+                commitPendingProposalsExpectation
+            ],
+            timeout: 5,
```

Comment:
perhaps 5 seconds is too long?

Reply:
done: de069d61502d413752b87fe9829a4ae8c3e65f07